src/esbmc/esbmc_parseoptions.cpp

```diff
@@ -1592,6 +1593,12 @@ bool esbmc_parseoptionst::process_goto_program(
       value_set_analysis.update(goto_functions);
     }
 
+    // unwind loops
+    optionst opts;
```

Comment:
Use the `no-unroll` option here to prevent this code from running, instead of in the class.